src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,17 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        Set<String> collect;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            collect = bufferedReader.lines()
+                    .flatMap(x -> Arrays.stream(x.toLowerCase()
+                            .replaceAll("[\\W\\d]", "")
+                            .split(""))).collect(Collectors.toSet());
+        } catch (IOException e) {
+            throw new FileNotFoundException("File not found!!!");
+        }
```

Comment:
you will wrap any IOException to FileNotFound ?We can just leave throws IOException

Reply:
done